handler/src/test/java/io/netty/handler/ssl/OpenSslEngineTest.java

```diff
@@ -131,8 +132,37 @@ public void testClientHostnameValidationFail() throws InterruptedException, SSLE
         super.testClientHostnameValidationFail();
     }
 
+    private static boolean isNpnSupported(String versionString) {
+        String[] versionStringParts = versionString.split(" ");
+        if (versionStringParts.length == 2 && "LibreSSL".equals(versionStringParts[0])) {
+            String[] versionParts = versionStringParts[1].split("\\.");
+            if (versionParts.length == 3) {
+                int major = Integer.parseInt(versionParts[0]);
+                if (major < 2) {
+                    return true;
+                }
+                if (major > 2) {
+                    return false;
+                }
+                int minor = Integer.parseInt(versionParts[1]);
+                if (minor < 6) {
+                    return true;
+                }
+                if (minor > 6) {
+                    return false;
+                }
+                int bugfix = Integer.parseInt(versionParts[2]);
+                if (bugfix > 0) {
```

Comment:
This looks wrong. Based on this, 2.6.0 would support NPN and 2.6.1 would not. This is probably why the `assumeFalse` is "working."

Reply:
@ejona86 actually they removed the supported in 2.6.1 so I think this is correct.

Reply:
O.o

Very well. :smile: